pkg/sql/compile/ddl.go

```diff
@@ -97,10 +97,11 @@ func (s *Scope) CreateTable(c *Compile) error {
 		if err != nil {
 			return err
 		}
-		if _, err := dbSource.Relation(c.ctx, def.Name); err != nil {
-			if err := dbSource.Create(c.ctx, def.Name, append(exeCols, exeDefs...)); err != nil {
-				return err
-			}
+		if _, err := dbSource.Relation(c.ctx, def.Name); err == nil {
```

Comment:
I don't know, but the right thing seems to be dbSource.Relation, then check err is table not exist.    You can have any strange errors such io error/network disconnect, which does not necessarily mean if the table exist or not.

Reply:
I can't understand what you want me to do

Reply:
dbSource.Relation, may return any error.   If you get some strange ones, ioError, your code will return table already exists.   The right thing seems to checking moerr.IsMoErrCode(err, TABLE_NOT_EXISTS) instead of err == nil

Seems both original code and your code treat any error as table not exists.   I don't think this is right.

Reply:
done

Reply:
Nasty.   Actually now it may not return ErrNotFound.   I think it could be ExpectedEOB.